doc/source/tune/_tutorials/tune-pytorch-lightning.rst

```diff
@@ -0,0 +1,220 @@
+.. _tune-pytorch-lightning:
+
+Using Tune to tune PyTorch Lightning models
+===========================================
+
+PyTorch Lightning is a framework which brings structure into training PyTorch models. It
+aims to avoid boilerplate code, so you don't have to write the same training
+loops all over again when building a new model.
+
+.. image:: /images/pytorch_lightning_full.png
+
+The main abstraction of PyTorch Lightning is the ``LightningModule`` class, which
+should be extended by your application. There is `a great post on how to transfer
+your models from vanilla PyTorch to Lightning <https://towardsdatascience.com/from-pytorch-to-pytorch-lightning-a-gentle-introduction-b371b7caaf09>`_.
+
+The class structure of PyTorch Lightning makes it very easy to define and tune model
+parameters. This tutorial will show you how to use Tune to find the best set of
+parameters for your application on the example of training a MNIST classifier. Notably,
+the ``LightningModule`` does not have to be altered at all for this - so you can
+use it plug and play for your existing models, assuming their parameters are configurable!
+
+.. note::
+
+    To run this example, you will need to install the following:
+
+    .. code-block:: bash
+
+        $ pip install ray torch torchvision pytorch-lightning
+
+PyTorch Lightning classifier for MNIST
+--------------------------------------
+Let's first start with the basic PyTorch Lightning implementation of an MNIST classifier.
+This classifier does not include any tuning code at this point.
+
+Our example builds on the MNIST example from the `blog post we talked about
+earlier <https://towardsdatascience.com/from-pytorch-to-pytorch-lightning-a-gentle-introduction-b371b7caaf09>`_.
+
+First, we run some imports
+
+.. literalinclude:: /../../python/ray/tune/examples/mnist_pytorch_lightning.py
+   :language: python
+   :start-after: __import_lightning_begin__
+   :end-before: __import_lightning_end__
+
+And then there is the Lightning model adapted from the blog post.
+Note that we left out the test set validation and made the model parameters
+configurable through a ``config`` dict that is passed on initialization.
+Also, we are able to specify a ``data_dir`` where the MNIST data will be stored.
+Lastly, we added a new metric, the validation accuracy, to the logs.
+
+.. literalinclude:: /../../python/ray/tune/examples/mnist_pytorch_lightning.py
+   :language: python
+   :start-after: __lightning_begin__
+   :end-before: __lightning_end__
+
+And that's it! You can now run ``train_mnist(config)`` to train the classifier, e.g.
+like so:
+
+.. code-block:: python
+
+    config = {
+        "layer_1_size": 128,
+        "layer_2_size": 256,
+        "lr": 1e-3,
+        "batch_size": 64
+    }
+    train_mnist(config)
+
+Tuning the model parameters
+---------------------------
```

Comment:
Consider breaking the below into subsubsections so that it is easily navigated.

```
Changing the CLI output
~~~~~~~~~~~~~~~~~~~~~~~
```
etc

Reply:
Good idea, I also restructured this a bit